### PR TITLE
feat(RichPresenceAssets): add YouTube and custom image support

### PR DESCRIPTION
**Pull request by Commandtechno**
**Please describe the changes this PR makes and why it should be merged:**
Added support for YouTube image URLs, and custom URLs in rich presence images similar to how the client does it.
You can now set the rich presence images as custom URLs which are prefixed with mp:

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: src/structures/Presence.js [hunk 1]
```diff
@@ -351,13 +351,21 @@ class RichPresenceAssets {
...
-        format,
-        size,
-      })
-    );
+    if (!this.smallImage) return null;
+    if (this.smallImage.includes(':')) {
+      const [platform, id] = this.smallImage.split(':');
+      switch (platform) {
+        case 'mp':
+          return `https://media.discordapp.net/${id}`;
```
Comment (vitorlops) [inline src/structures/Presence.js:359]:
```suggestion
          return `https://cdn.discordapp.com/${id}`;
```

File: src/structures/Presence.js [hunk 2]
```diff
@@ -367,11 +375,22 @@ class RichPresenceAssets {
...
     if (!this.largeImage) return null;
-    if (/^spotify:/.test(this.largeImage)) {
-      return `https://i.scdn.co/image/${this.largeImage.slice(8)}`;
-    } else if (/^twitch:/.test(this.largeImage)) {
-      return `https://static-cdn.jtvnw.net/previews-ttv/live_user_${this.largeImage.slice(7)}.png`;
+    if (this.largeImage.includes(':')) {
+      const [platform, id] = this.largeImage.split(':');
+      switch (platform) {
+        case 'mp':
+          return `https://media.discordapp.net/${id}`;
```
Comment (vitorlops) [inline src/structures/Presence.js:382]:
```suggestion
          return `https://cdn.discordapp.com/${id}`;
```

**Review by vitorlops — COMMENTED** [review]
Perhaps using `cdn` URL instead of `media`?

**Comment by Commandtechno** [comment]
`mp` stands for media proxy thats why I went with it, but after testing, cdn.discordapp.com doesn't seem to work, I might also be missing something.

What I could do is extract the original URL from the media proxy URL, but I don't see anything wrong with the media proxy url either.
, but I don't see anything wrong with the media proxy url either.
Media proxy URL: https://media.discordapp.net/external/0g8mlaBL2Xax0nf8f1WjenSR9-SL4x98RSze3wxnF8I/https/cdn.discordapp.com/emojis/765141043603046420.gif

CDN URL: https://cdn.discordapp.com/external/0g8mlaBL2Xax0nf8f1WjenSR9-SL4x98RSze3wxnF8I/https/cdn.discordapp.com/emojis/765141043603046420.gif

**Review by DraftProducts — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by thebigwolfy — APPROVED** [review]

**Comment by HitmanPhoenix18** [comment]
Hey there, sorry i haven't responded in a while!! I was grounded until I
got my act together (lmao)!
So I was just done trying to find the problems with the codes and my uncle
said that he might be able to help you guys!!! For he is a really good
coder! So please start using my og email @***@***.***
***@***.***> and thank you so much! I will talk to my uncle
about it if you guys want the help!

**Pull request merged**